core/src/main/java/io/jenkins/pluginhealth/scoring/utility/GitHubWorkflowReader.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2023 Jenkins Infra
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package io.jenkins.pluginhealth.scoring.utility;
+
+import java.nio.file.Path;
+
+import io.jenkins.pluginhealth.scoring.probes.Probe;
+
+public abstract class GitHubWorkflowReader extends Probe {
+    public abstract String getWorkflowDefinition(Path githubWorkflow);
```

Comment:
This shouldn't take a parameter and have a javadoc.
```suggestion
    /**
     * Returns the path to the GitHub Workflow definition which should be use in one of the actions of the plugin repository.
     *
     * @return the workflow definition used in one of the jobs of one of the actions of the plugin repository.
     */
    public abstract String getWorkflowDefinition();
```